### PR TITLE
[DPTP-1796] Add FIPS to ipi-install chain and introduce the fips-check step

### PR DESCRIPTION
**Pull request by droslean**
In this PR
- Adds a new step that injects the fips field in install-config
- Adds the fips-check step that is responsible to detect if the nodes of the cluster have FIPS enabled.
- Changes all the fips related presubmits to use those steps instead.

follow-up: There are some handcrafted periodics that need to be changed as well.  

Signed-off-by: Nikolaos Moraitis <nmoraiti@redhat.com>

### CODE REVIEW

File: ci-operator/config/openshift/installer/openshift-installer-master.yaml
Comment (petr-muller) [inline ci-operator/config/openshift/installer/openshift-installer-master.yaml:391]: The tests should probably not be in a `pre:` phase.

File: ci-operator/step-registry/fips-check/fips-check-ref.yaml
Comment (petr-muller) [inline ci-operator/step-registry/fips-check/fips-check-ref.yaml:10]: Should this have a default?

File: ci-operator/step-registry/fips-check/fips-check-ref.yaml
Comment (petr-muller) [inline ci-operator/step-registry/fips-check/fips-check-ref.yaml:12]: Just wondering - should this step be done pretty much everywhere, at least in the workflows that run the shared e2e testsuite?

File: ci-operator/step-registry/ipi/conf/fips/ipi-conf-fips-commands.sh
Comment (petr-muller) [inline ci-operator/step-registry/ipi/conf/fips/ipi-conf-fips-commands.sh:9]: I don't think this needs to be a separate step -- I remember Bruno suggested this can simply be done in https://steps.ci.openshift.org/reference/ipi-conf - it can just have an additional param and that's it.

Comment (droslean) [reply ci-operator/step-registry/ipi/conf/fips/ipi-conf-fips-commands.sh:9]: It seems that this is the practice that its already used: Check for example [here](https://github.com/openshift/release/blob/master/ci-operator/step-registry/ovn/conf/ovn-conf-commands.sh)

Comment (droslean) [reply ci-operator/step-registry/ipi/conf/fips/ipi-conf-fips-commands.sh:9]: But on a second thought, I think it is better to just put it in `ipi-conf` as you suggested.

File: ci-operator/config/openshift/installer/openshift-installer-master.yaml
Comment (petr-muller) [inline ci-operator/config/openshift/installer/openshift-installer-master.yaml:404]: Technically this is even the `openshift-e2e-test` wortkflow and you do not need to override its `test`, just `pre`.

/hold

**Review by petr-muller — APPROVED** [review]
The test configs can be simplified even more but that's just a minor detail, I'll leave it up to you whether you want to address it or not (holding, feel free to unhold)

**Pull request merged**
